pkg/filter/pipelinerun.go

```diff
@@ -0,0 +1,156 @@
+	TektonPipelineRunTriggeredBuilds = fmt.Sprintf("%s/pipelinerun-triggered-builds", Prefix)
+)
+
+// searchBuildRunForRunOwner inspect the object owners for Tekton Run and returns it, otherwise nil.
+func searchBuildRunForRunOwner(br *v1alpha1.BuildRun) *types.NamespacedName {
+	for _, ownerRef := range br.OwnerReferences {
+		if ownerRef.APIVersion == constants.TektonAPIv1alpha1 && ownerRef.Kind == "Run" {
+			return &types.NamespacedName{Namespace: br.GetNamespace(), Name: ownerRef.Name}
+		}
+	}
+	return nil
+}
+
+// filterBuildRunOwnedByRun filter out BuildRuns objects not owned by Tekton Run.
+func filterBuildRunOwnedByRun(obj interface{}) bool {
+	br, ok := obj.(*v1alpha1.BuildRun)
+	if !ok {
+		return false
+	}
+	return searchBuildRunForRunOwner(br) != nil
+}
+
+// pipelineRunReferencesShipwright checks if the informed PipelineRun is reffering to a Shipwright
+// resource via TaskRef.
+func pipelineRunReferencesShipwright(pipelineRun *tknv1beta1.PipelineRun) bool {
+	if pipelineRun.Status.PipelineSpec == nil {
+		return false
+	}
+	for _, task := range pipelineRun.Status.PipelineSpec.Tasks {
+		if task.TaskRef == nil {
+			continue
+		}
+		if task.TaskRef.APIVersion == constants.ShipwrightAPIVersion {
+			return true
+		}
+	}
+	return false
+}
+
+// EventFilterPredicate predicate filter for the basic inspections in the object, filtering only what
+// needs to go through reconciliation. PipelineRun objects referencing Custom-Tasks are also skipped.
+func EventFilterPredicate(obj client.Object) bool {
+	logger := logr.New(log.Log.GetSink()).
+		WithValues("namespace", obj.GetNamespace(), "name", obj.GetName())
+
+	pipelineRun, ok := obj.(*tknv1beta1.PipelineRun)
+	if !ok {
+		logger.V(0).Error(nil, "Unable to cast object as Tekton PipelineRun")
+		return false
+	}
+
+	if !pipelineRun.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.V(0).Info("Marked for deletion")
+		return false
+	}
+
+	if pipelineRun.Spec.PipelineRef == nil {
+		logger.V(0).Info("Skipping due nil .Spec.PipelineRef")
+		return false
+	}
+
+	if pipelineRun.Status.PipelineSpec == nil {
+		logger.V(0).Info("Skipping due to nil .Status.PipelineSpec")
+		return false
+	}
+
+	if pipelineRunReferencesShipwright(pipelineRun) {
+		logger.V(0).Info("Skipping due to being part of a Custom-Task")
+		return false
+	}
+	return true
+}
+
+// ParsePipelineRunStatus parte the informed object status to extract its status.
+func ParsePipelineRunStatus(
+	ctx context.Context,
+	now time.Time,
+	pipelineRun *tknv1beta1.PipelineRun,
+) (string, error) {
+	switch {
+	case pipelineRun.IsDone():
+		if pipelineRun.Status.GetCondition(apis.ConditionSucceeded).IsTrue() {
+			return tknv1beta1.PipelineRunReasonSuccessful.String(), nil
+		}
+		return tknv1beta1.PipelineRunReasonFailed.String(), nil
+	case pipelineRun.IsCancelled():
+		return tknv1beta1.PipelineRunReasonCancelled.String(), nil
+	case pipelineRun.HasTimedOut(ctx, clock.NewFakePassiveClock(now)):
```

Comment:
Similar as above, this causes a calculation to happen but does not check whether the status of the PipelineRun really is representing a timeout.

Reply:
Well, taking in consideration my [last comment](https://github.com/shipwright-io/triggers/pull/4#discussion_r1033385400), I think we need to rely on what Tekton provides. Do you have a suggestion of an alternative path using what their package provides?

Reply:
I propose to look at the condition with `type=Succeeded`. For final states such as canceled, or timed out, we should check that `status` is `"False"` and check for the expected reason. They are documented here: https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#monitoring-execution-status

Reply:
I would not necessarily rely on their functions as I personally consider them less API-stable than the documented reasons.

Reply:
> I would not necessarily rely on their functions as I personally consider them less API-stable than the documented reasons.

At least on the initial project state, I think we should follow Tekton's upstream and improve that part as we need. Right from the start I'm reluctant to deviate from Tekton upstream, so let's add a new issue and follow up in the next interactions on this project?